x/snapshot/exported/types.go

```diff
@@ -36,6 +29,39 @@ type ValidatorI interface {
 	IsBonded() bool                        // whether the validator is bonded
 }
 
+// QuadraticWeightFunc returns floor(sqrt(consensusPower)) as the weight
+func QuadraticWeightFunc(consensusPower sdk.Uint) sdk.Uint {
+	bigInt := consensusPower.BigInt()
+
+	return sdk.NewUintFromBigInt(bigInt.Sqrt(bigInt))
+}
+
+// Slasher provides necessary functions to the validator information
+type Slasher interface {
+	IsTombstoned(ctx sdk.Context, consAddr sdk.ConsAddress) bool // whether a validator is tombstoned
+}
+
+// IsTombstoned returns a function that checks if validator is tombstoned
+func IsTombstoned(ctx sdk.Context, slasher Slasher) func(ValidatorI) bool {
+	return func(v ValidatorI) bool {
+		consAdd, err := v.GetConsAddr()
+		if err != nil {
+			return true
+		}
+
+		return slasher.IsTombstoned(ctx, consAdd)
+	}
+}
```

Comment:
this doesn't make much sense to me. There is not a lot of value putting this here and it has a very confusing signature